pkg/appmanager/appManager.go

```diff
@@ -1110,20 +1110,28 @@ func (appMgr *Manager) syncRoutes(
 		if route.ObjectMeta.Namespace != sKey.Namespace {
 			continue
 		}
+
+		//FIXME(kenr): why do we process services that aren't associated
+		//             with a route?
+		svcName := getRouteCanonicalService(route)
+		if existsRouteServiceName(route, sKey.ServiceName) {
+			svcName = sKey.ServiceName
+		}
+
 		if nil != route.Spec.TLS {
 			switch route.Spec.TLS.Termination {
 			case routeapi.TLSTerminationPassthrough:
-				updateDataGroupForPassthroughRoute(route, DEFAULT_PARTITION,
+				updateDataGroupForPassthroughRoute(route, svcName, DEFAULT_PARTITION,
```

Comment:
There is a problem with updating the data groups for one route with multiple services (which is what this commit is introducing). The problem is that the key in the data group is the host name defined in the route and the value is the pool name. The key is unique, so in this case the data group is updated for the same host multiple times - once for each backend, and the last update 'wins'.

Reply:
For this commit, I will only update for the main (canonical) service in the route since this change is not doing A/B yet (just creating the extra pools).